File: Domasno3/infobank-frontend/src/app/models/BankEntity.ts
export class BankEntity {
  public _id: string;
  public _class: string;
  public lat: number;
  public lon: number;
  public name: string;
  public type: string;
  public details:  { [key: string]: string } = {};

  constructor(id: string, _class: string, lat: number, lon: number, name: string, type: string, details: {}) {
    this._id = id;
    this._class = _class;
    this.lat = lat;
    this.lon = lon;
    this.name = name;
    this.type = type;
    this.details = details;
  }
}

File: infobank-backend/Navigation/src/main/java/mk/finki/ukim/navigation/Model/BankEntity.java
package mk.finki.ukim.navigation.Model;

import lombok.Data;
import lombok.NoArgsConstructor;
import mk.finki.ukim.navigation.Model.enumerations.BankType;
import org.springframework.data.annotation.Id;
import javax.annotation.PostConstruct;
import java.util.HashMap;
import java.util.Map;

@Data
@NoArgsConstructor
public class BankEntity extends LocationInfo {

    @Id
    private String id;
    private BankType type;
    private String name;
    private Map<String, String> details;

    @PostConstruct
    private void postConstruct() {
        details = new HashMap<>();
    }

    public BankEntity(Map<String, String> data) {
        HashMap<String, String> detailsCopy = new HashMap<>(data);
        id = detailsCopy.remove("id");
        name = detailsCopy.remove("name");
        type = BankType.valueOf(detailsCopy.remove("amenity"));
        lon = Double.parseDouble(detailsCopy.get("lon"));
        lat = Double.parseDouble(detailsCopy.get("lat"));
        details = detailsCopy;
        detailsCopy.remove("lat");
        detailsCopy.remove("lon");
    }
}
